### PR TITLE
ONNX FE handles old API extensions

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - *Fix custom op test for ONNX frontend*
 - *ONNX frontend use legacy conversion extension for old API extensions*

### Tickets:
 - *89374*

### CODE REVIEW

File: src/core/include/openvino/core/op_extension.hpp [hunk 1]
```diff
@@ -100,4 +100,6 @@ class OpExtension : public BaseOpExtension {
     }
 };
 
+/** @brief Class to distinguish legacy extension. */
+class LegacyOpExtension : public BaseOpExtension {};
```
Comment (ilyachur) [inline src/core/include/openvino/core/op_extension.hpp:104]: Can we move this class to private or developer API?

Comment (praasz) [reply src/core/include/openvino/core/op_extension.hpp:104]: I think, yes. 
Should be done by some macro or moved to `detail` namespace?

Comment (ilyachur) [reply src/core/include/openvino/core/op_extension.hpp:104]: I think we can just move it to the same place which is used for SOExtension for example.

Comment (praasz) [reply src/core/include/openvino/core/op_extension.hpp]: Done.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**